promql/engine.go

```diff
@@ -2993,11 +2985,13 @@ func unwrapParenExpr(e *parser.Expr) {
 	}
 }
 
-func unwrapStepInvariantExpr(e parser.Expr) parser.Expr {
-	if p, ok := e.(*parser.StepInvariantExpr); ok {
+func unwrapStepInvariantAndParenExprs(e *parser.Expr) parser.Expr {
+	unwrapParenExpr(e)
+	if p, ok := (*e).(*parser.StepInvariantExpr); ok {
+		unwrapParenExpr(&p.Expr)
```

Comment:
This is a change in behaviour: previously the second `unwrapParenExpr` would be called on a local variable.

Maybe it's ok, since the previous code modified `p.Expr` in the first call, but if it's ok I would like this PR to explain why.

Reply:
> Previously, the 2nd unwrapParenExpr would be called on a pointer to local variable, which had the value p.Expr. Thus, effectively, it was called on &p.Expr

I don't think so.  Changing a local variable does not change `p.Expr`.
Can you follow it through in a debugger and show what the addresses are, if you still think you are correct?